Gaiad export fails with nil pointer dereference

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺ 
v                            ✰  Thanks for opening an issue! ✰    
v    Before smashing the submit button please review the template.
v    Please also ensure that this is not a duplicate issue :)  
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  --> 

## Summary of Bug 
<!-- Concisely describe the issue --> 
`gaiad export` on v0.22.0 / gaia-7001 fails with the following error.
```
root@gaia:~# gaiad export
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x9a2e6e]

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).NewContext(0xc42047a180, 0xfd6501, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /root/go/src/github.com/cosmos/cosmos-sdk/baseapp/baseapp.go:232 +0x5e
github.com/cosmos/cosmos-sdk/cmd/gaia/app.(*GaiaApp).ExportAppStateAndValidators(0xc4200e3ba0, 0xc420a31660, 0xfd85a0, 0xc42000e090, 0x0, 0x0, 0x0, 0x0, 0x0)
        /root/go/src/github.com/cosmos/cosmos-sdk/cmd/gaia/app/app.go:184 +0x6a
main.exportAppStateAndTMValidators(0xfcff00, 0xc420a31660, 0xfd85a0, 0xc42000e090, 0x0, 0x0, 0x0, 0x0, 0xc420aeba88, 0xa23437, ...)
        /root/go/src/github.com/cosmos/cosmos-sdk/cmd/gaia/cmd/gaiad/main.go:53 +0x86
github.com/cosmos/cosmos-sdk/server.ConstructAppExporter.func1(0xc4209555f0, 0xc, 0xfcff00, 0xc420a31660, 0x0, 0x0, 0x997550, 0xcc61c0, 0x0, 0xe15c8e, ...)
        /root/go/src/github.com/cosmos/cosmos-sdk/server/constructors.go:82 +0x14c
github.com/cosmos/cosmos-sdk/server.ExportCmd.func1(0xc420aa2900, 0x1709808, 0x0, 0x0, 0x0, 0x0)
        /root/go/src/github.com/cosmos/cosmos-sdk/server/export.go:23 +0xf9
github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra.(*Command).execute(0xc420aa2900, 0x1709808, 0x0, 0x0, 0xc420aa2900, 0x1709808)
        /root/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra/command.go:698 +0x46d
github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200c9440, 0xcfa2c0, 0xc4200f3e01, 0xc420a313a0)
        /root/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra/command.go:783 +0x2e4
github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200c9440, 0xc420a313a0, 0xc4200f3ed8)
        /root/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra/command.go:736 +0x2b
github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/libs/cli.Executor.Execute(0xc4200c9440, 0xf53810, 0x2, 0xc4209555f0)
        /root/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/libs/cli/setup.go:89 +0x4e
main.main()
        /root/go/src/github.com/cosmos/cosmos-sdk/cmd/gaia/cmd/gaiad/main.go:38 +0x214

```
After talking with @AdityaSripal we've traced the issue to `app.setCheckState` not being called on the gaia app that is created for export. This can be fixed by adding that call to the end of `app.initFromStore`, will submit a PR to that effect.
## Steps to Reproduce
<!-- What commands in order should someone run to reproduce your problem --> 
Call `gaiad export` with gaiad on v0.22.0
____
#### For Admin Use
 - [ ] Not duplicate issue
 - [ ] Appropriate labels applied
 - [ ] Appropriate contributors tagged
 - [ ] Contributor assigned/self-assigned

--- a/PENDING.md
+++ b/PENDING.md
@@ -52,3 +52,4 @@ BUG FIXES
 *  \#1797 Fix off-by-one error in slashing for downtime
 *  \#1787 Fixed bug where Tally fails due to revoked/unbonding validator
 *  \#1766 Fixes bad example for keybase identity
+*  \#1799 Fix `gaiad export`

--- a/baseapp/baseapp.go
+++ b/baseapp/baseapp.go
@@ -190,7 +190,7 @@ func (app *BaseApp) initFromStore(mainKey sdk.StoreKey) error {
 	if main == nil {
 		return errors.New("baseapp expects MultiStore with 'main' KVStore")
 	}
-
+	app.setCheckState(abci.Header{})
 	return nil
 }
 

--- a/baseapp/baseapp.go
+++ b/baseapp/baseapp.go
@@ -190,6 +190,7 @@ func (app *BaseApp) initFromStore(mainKey sdk.StoreKey) error {
 	if main == nil {
 		return errors.New("baseapp expects MultiStore with 'main' KVStore")
 	}
+	// Needed for `gaiad export`, which inits from store but never calls initchain
 	app.setCheckState(abci.Header{})
 	return nil
 }

--- a/cmd/gaia/app/app_test.go
+++ b/cmd/gaia/app/app_test.go
@@ -1,9 +1,15 @@
 package app
 
 import (
+	"os"
+	"testing"
+
 	"github.com/cosmos/cosmos-sdk/wire"
 	"github.com/cosmos/cosmos-sdk/x/auth"
 	"github.com/cosmos/cosmos-sdk/x/stake"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/db"
+	"github.com/tendermint/tendermint/libs/log"
 
 	abci "github.com/tendermint/tendermint/abci/types"
 )
@@ -30,4 +36,15 @@ func setGenesis(gapp *GaiaApp, accs ...*auth.BaseAccount) error {
 	gapp.Commit()
 
 	return nil
+}
+
+func TestGaiadExport(t *testing.T) {
+	db := db.NewMemDB()
+	gapp := NewGaiaApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil)
+	setGenesis(gapp)
+
+	// Making a new app object with the db, so that initchain hasn't been called
+	newGapp := NewGaiaApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil)
+	_, _, err := newGapp.ExportAppStateAndValidators()
+	require.NoError(t, err, "ExportAppStateAndValidators should not have an error")
 }

